docs/source/reference/envvars.rst

```diff
@@ -467,6 +467,16 @@ GPU support
    Strict strides checking is deprecated and may be removed in future. See
    :ref:`deprecation-strict-strides`.
 
+.. envvar:: NUMBA_CUDA_LOW_OCCUPANCY_WARNINGS
+
+   Enable warnings if the grid size is too small relative to the number of
+   steraming multiprocessors (SM). This option is on by default (1).
+
+   The heuristic checked is whether gridsize < 2 * (number of SMs). NOTE: The absence of
+   a warning does not imply a good gridsize relative to the number of SMs. Disabling
+   this warning will reduce the number of CUDA API calls, as the heuristic needs to
+   check the number of SMs for the device in the current context.
```

Comment:
Does lead to a noticeable compilation speed gain?

Reply:
This is for launch speed rather than compilation speed. In the past there have been performance issues due to excessive getting of contexts, e.g.: https://github.com/numba/numba/pull/5189 - so I think if we add a get of the context in the launch pipeline, it's good to have a way around it as well.